lib/publiccloud/azure.pm

```diff
@@ -201,16 +237,25 @@ sub upload_img {
     ## 1. Ensure the image definition in the Azure Compute Gallery exists
     ## 2. Create a new image version for that definition. Use the link to the uploaded blob to create this version
 
+
     # Print image definitions as a help to debug possible conflicting definitions
     my $images = script_output("az sig image-definition list -g '$resource_group' -r '$gallery'");
-    record_info("img-def", "Existing image definitions:\n$images");
-
-    my $arch = (get_var('PUBLIC_CLOUD_ARCH', 'x86_64'));
-    $arch = $arch eq 'x86_64' ? 'x64' : 'Arm64';
+    record_info("img-defs", "Existing image definitions:\n$images");
+    my $arch = (get_var('PUBLIC_CLOUD_ARCH', 'x86_64')) eq 'x86_64' ? 'x64' : 'Arm64';
+
+    # Create new definition only, if there is no previous definition present
+    my $definition = $self->get_image_definition($resource_group, $gallery, $publisher, $offer, $sku);
+    if (defined $definition) {
+        record_info("use img-def", "Using found image definitions:\n$definition");
+    } else {
+        $definition = $self->generate_azure_image_definition();
+        my $gen = ($sku =~ "gen2" ? "V2" : "V1");
```

Comment:
so `'aarch64-gen2'` will give us `V1` ? :thinking:

Reply:
`=~` but I know, perl is hard sometimes :wink: